### PR TITLE
Reduce code duplication in ExpressionUtil

### PR DESCRIPTION
**Pull request by k-jamroz**
Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/ExpressionUtil.java [hunk 1]
```diff
@@ -153,33 +153,15 @@ public static FunctionEx<JetSqlRow, JetSqlRow> projectionFn(
...
-                Object[] result = new Object[projections.size()];
-                for (int i = 0; i < projections.size(); i++) {
-                    result[i] = evaluate(projections.get(i), row, context);
-                }
-                return new JetSqlRow(context.getSerializationService(), result);
-            } else {
-                return null;
-            }
-        };
+        return row0 -> projection(predicate, projections, row0.getRow(), context);
```
Comment (k-jamroz) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/ExpressionUtil.java:164]: there is additional null-comparison in this variant. Do we care?

File: hazelcast/src/main/java/com/hazelcast/sql/impl/row/JetSqlRow.java [hunk 2]
```diff
@@ -81,6 +81,7 @@ public SerializationService getSerializationService() {
         return ss;
     }
 
+    @Nonnull
```
Comment (viliam-durina) [inline hazelcast/src/main/java/com/hazelcast/sql/impl/row/JetSqlRow.java:84]: Why was this removed?

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/generator/CounterRow.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+package com.hazelcast.jet.sql.impl.connector.generator;
+
+import com.hazelcast.sql.impl.row.Row;
+
+import java.util.Objects;
+
+/**
+ * Row with single integer field
+ */
+class CounterRow implements Row {
```
Comment (viliam-durina) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/generator/CounterRow.java:26]: Why "counter" in the name? It's just a value here, not a counter.

Also, in contract to the long variant, this should contain `Int` in the name:
```suggestion
class SingleIntRow implements Row {
```

Comment (k-jamroz) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/generator/CounterRow.java:26]: It is only internal class, used as counter, but I like your name better

File: hazelcast/src/main/java/com/hazelcast/sql/impl/row/JetSqlRow.java (hunk 2 above)
Comment (k-jamroz) [reply hazelcast/src/main/java/com/hazelcast/sql/impl/row/JetSqlRow.java:84]: do you mean "added"? I wanted to be explicit with regard to (non)nullability. Some functions in `ExpressionUtils` return null `JetSqlRow`

Comment (viliam-durina) [reply hazelcast/src/main/java/com/hazelcast/sql/impl/row/JetSqlRow.java:84]: Oh, my bad :laughing:

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/generator/CounterRow.java (hunk 3 above)
Comment (viliam-durina) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/generator/CounterRow.java]: Even the name of a local variable matters, it should ideally say what the value is.

**Review by viliam-durina — APPROVED** [review]

**Review by Fly-Style — APPROVED** [review]
Great job, thank you! LGTM.
